dds/idl/annotations.h

```diff
@@ -345,6 +345,7 @@ namespace OpenDDS {
     bool xcdr1;
     bool xcdr2;
     bool xml;
+    bool unaligned;
```

Comment:
`UNALIGNED_CDR` also needs to be added to `@OpenDDS::data_representation` or else this doesn't do much: https://github.com/objectcomputing/OpenDDS/blob/fe59c08bc4316281a4b7805d679327435ff7ac8a/dds/idl/annotations.cpp#L374

https://github.com/objectcomputing/OpenDDS/blob/fe59c08bc4316281a4b7805d679327435ff7ac8a/dds/idl/annotations.cpp#L419-L421

Reply:
The goal is to keep Unaligned as the interpreted default for transports besides rtps_udp.  This is to maintain compatibility with earlier OpenDDS versions.  I don't think we want the users explicitly setting unaligned in idl. The long term goal is removal of Unaligned, and we would need to consider a lot more cases if users could set it.